api/v1beta1/metal3clustertemplate_webhook_test.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestMetal3ClusterTemplateDefault(t *testing.T) {
+	g := NewWithT(t)
+
+	c := &Metal3ClusterTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+		},
+		Spec: Metal3ClusterTemplateSpec{},
+	}
+	c.Default()
+
+	g.Expect(c.Spec).To(Equal(Metal3ClusterTemplateSpec{}))
+}
+
+func TestMetal3ClusterTemplateValidation(t *testing.T) {
```

Comment:
Could you add one test for when the template is invalid also?

Reply:
Done :)